alerting/src/main/kotlin/org/opensearch/alerting/PPLSQLMonitorRunner.kt

```diff
@@ -0,0 +1,576 @@
+import org.opensearch.alerting.PPLUtils.executePplQuery
+import org.opensearch.alerting.PPLUtils.findEvalResultVar
+import org.opensearch.alerting.PPLUtils.findEvalResultVarIdxInSchema
+import org.opensearch.alerting.alertsv2.AlertV2Indices
+import org.opensearch.alerting.modelv2.AlertV2
+import org.opensearch.alerting.modelv2.MonitorV2
+import org.opensearch.alerting.modelv2.MonitorV2RunResult
+import org.opensearch.alerting.modelv2.PPLSQLMonitor
+import org.opensearch.alerting.modelv2.PPLSQLMonitorRunResult
+import org.opensearch.alerting.modelv2.PPLSQLTrigger
+import org.opensearch.alerting.modelv2.PPLSQLTrigger.ConditionType
+import org.opensearch.alerting.modelv2.PPLSQLTrigger.NumResultsCondition
+import org.opensearch.alerting.modelv2.PPLSQLTrigger.TriggerMode
+import org.opensearch.alerting.modelv2.PPLSQLTriggerRunResult
+import org.opensearch.alerting.modelv2.TriggerV2.Severity
+import org.opensearch.alerting.opensearchapi.InjectorContextElement
+import org.opensearch.alerting.opensearchapi.retry
+import org.opensearch.alerting.opensearchapi.suspendUntil
+import org.opensearch.alerting.opensearchapi.withClosableContext
+import org.opensearch.alerting.script.PPLTriggerExecutionContext
+import org.opensearch.alerting.settings.AlertingSettings
+import org.opensearch.common.xcontent.XContentFactory
+import org.opensearch.commons.alerting.alerts.AlertError
+import org.opensearch.commons.alerting.model.Alert
+import org.opensearch.commons.alerting.model.ScheduledJob.Companion.SCHEDULED_JOBS_INDEX
+import org.opensearch.commons.alerting.model.action.Action
+import org.opensearch.commons.alerting.model.userErrorMessage
+import org.opensearch.core.common.Strings
+import org.opensearch.core.rest.RestStatus
+import org.opensearch.core.xcontent.ToXContent
+import org.opensearch.transport.TransportService
+import org.opensearch.transport.client.node.NodeClient
+import java.time.Instant
+import java.time.ZoneOffset.UTC
+import java.time.format.DateTimeFormatter
+import java.time.temporal.ChronoUnit
+import java.util.Locale
+import kotlin.math.min
+
+object PPLSQLMonitorRunner : MonitorV2Runner() {
+    private val logger = LogManager.getLogger(javaClass)
+
+    override suspend fun runMonitorV2(
+        monitorV2: MonitorV2,
+        monitorCtx: MonitorRunnerExecutionContext, // MonitorV2 reads from same context as Monitor
+        periodEnd: Instant,
+        dryRun: Boolean,
+        manual: Boolean,
+        executionId: String,
+        transportService: TransportService,
+    ): MonitorV2RunResult<*> {
+        if (monitorV2 !is PPLSQLMonitor) {
+            throw IllegalStateException("Unexpected monitor type: ${monitorV2.javaClass.name}")
+        }
+
+        if (monitorV2.id == MonitorV2.NO_ID) {
+            throw IllegalStateException("Received PPL Monitor to execute that unexpectedly has no ID")
+        }
+
+        logger.debug("Running PPL Monitor: ${monitorV2.name}. Thread: ${Thread.currentThread().name}")
```

Comment:
logging monitor id is more pertinent for debugging since name need not be unique imo